src/server/snapshot.h

```diff
@@ -171,6 +172,8 @@ class SliceSnapshot {
     size_t savecb_calls = 0;
     size_t keys_total = 0;
   } stats_;
+
+  util::fb2::Mutex bucket_ser_mu_;
```

Comment:
you have different mutex one for snapshot and one for cluster, I think this is a problem.
If we have a snapshot created at the same time there is a migration created and we yield in the bucket serialization